### PR TITLE
MBS-13224: Show edit/note restrictions from user profile

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-13224

# Problem
We have for a long time chosen not to show whether a user was blocked from editing or leaving edit notes publicly in order to avoid publicly shaming editors.
Sadly, this means already banned editors get reported again, kind of wasting the time of both re-reporters and admins.
Additionally, it's unfair to the user *not* to display this info, since other editors can demand the user react to their edit notes without knowing they cannot.

# Solution
This shows active editing/voting an edit note restrictions for the user to other logged in users only and only when they visit their profile, which is the least shaming way of doing it but still visible if someone is thinking of re-reporting.

The "Untrusted" restriction is explicitly not shown to other users. "Untrusted" is usually set as a "hidden" "minor" restriction for users who just need an extra eye on their edits, and we don't even notify the restricted user of it, so it wouldn't make sense to notify others.

# Testing
Manually, plus added a test for this (and while at it turned another pointless test for this page into one that does something, on a separate commit).

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, retest this please

**Review by mwiencek — APPROVED** [review]
Tested locally and works well. I wonder if it'd make more sense to move this info underneath/closer to the "Report this user for bad behavior" link? That way it's less prominent if you're just looking at the profile info, but more prominent if you're looking to report the user?

**Comment by reosarevok** [comment]
Hmm. The only issue with that is it being less visible for the "why is this person not answering my notes" use case :) But maybe? @Aerozol: opinions? Currently looks like this:

image_2023-09-05_213116964

**Comment by yvanzo** [comment]
> visible for the "why is this person not answering my notes" use case

It seems a pretty common use case to care about indeed. We can keep this info under “General information” but also add a hint next to the report link. For example:

    Report this user for bad behavior (despite the above active restrictions)

**Comment by Aerozol** [comment]
It looks fine to me (nitpick, I think ‘Active restrictions’ could be cut down to ‘Restrictions’).

I would probably advise against adding more text to ‘Report this user for bad behaviour’, if we don’t have to. Perhaps if someone has active restrictions we can add an asterisk, or a [?] box with a mouseover saying ‘user has current restrictions, see above’. But perhaps that’s best left for a follow-up ticket if reo finds that users are often reported repeatedly, after already being restricted.

File: root/user/UserProfile.js [hunk 1]
```diff
@@ -277,6 +280,13 @@ const UserProfileInformation = ({
           {memberSince}
         </UserProfileProperty>
 
+        {/* Only relevant to logged in users, avoid more public shame */}
+        {viewingUser && restrictions.length ? (
+          <UserProfileProperty name={l(addColonText('Active restrictions'))}>
```
Comment (yvanzo) [inline root/user/UserProfile.js:285]: Suggestion by @Aerozol in an actionnable way:
```suggestion
          <UserProfileProperty name={l(addColonText('Restrictions'))}>
```

**Comment by yvanzo** [comment]
Applied @Aerozol’s suggestion to cut down the wording.

Opened the follow-up ticket MBS-13267 for adding reminders when reporting a user that has restrictions, so we can merge the current changes already.

**Pull request merged**